cmd/ftl-provisioner-cloudformation/status.go

```diff
@@ -86,49 +93,110 @@ func failure(stack *types.Stack) (*connect.Response[provisioner.StatusResponse],
+	}
+	return m, nil
+}
+
+func (c *CloudformationProvisioner) updateResources(ctx context.Context, outputs []types.Output, update []*provisioner.Resource) error {
+	byResourceID, err := outputsByResourceID(outputs)
+	if err != nil {
+		return fmt.Errorf("failed to group outputs by resource ID: %w", err)
+	}
+
+	for _, resource := range update {
+		if postgres, ok := resource.Resource.(*provisioner.Resource_Postgres); ok {
+			if postgres.Postgres == nil {
+				postgres.Postgres = &provisioner.PostgresResource{}
+			}
+			if postgres.Postgres.Output == nil {
+				postgres.Postgres.Output = &provisioner.PostgresResource_PostgresResourceOutput{}
+			}
+
+			if err := c.updatePostgresOutputs(ctx, postgres.Postgres.Output, resource.ResourceId, byResourceID[resource.ResourceId]); err != nil {
+				return fmt.Errorf("failed to update postgres outputs: %w", err)
 			}
+		} else if _, ok := resource.Resource.(*provisioner.Resource_Mysql); ok {
+			panic("mysql not implemented")
 		}
 	}
 	return nil
 }
 
-func endpointToDSN(endpoint, database string, port int) string {
-	return fmt.Sprintf("postgres://%s:%d/%s?user=postgres&password=password", endpoint, port, database)
+func (c *CloudformationProvisioner) updatePostgresOutputs(ctx context.Context, to *provisioner.PostgresResource_PostgresResourceOutput, resourceID string, outputs []types.Output) error {
+	byName, err := outputsByPropertyName(outputs)
+	if err != nil {
+		return fmt.Errorf("failed to group outputs by property name: %w", err)
+	}
+
+	fmt.Fprintf(os.Stderr, "byName: %v\n", byName)
+
+	// TODO: Move to provisioner workflow
+	secretARN := *byName[PropertyMasterUserARN].OutputValue
+	username, password, err := c.secretARNToUsernamePassword(ctx, secretARN)
+	if err != nil {
+		return fmt.Errorf("failed to get username and password from secret ARN: %w", err)
+	}
+
+	to.ReadDsn = endpointToDSN(*byName[PropertyDBReadEndpoint].OutputValue, resourceID, 5432, username, password)
+	to.WriteDsn = endpointToDSN(*byName[PropertyDBWriteEndpoint].OutputValue, resourceID, 5432, username, password)
+	adminEndpoint := endpointToDSN(*byName[PropertyDBReadEndpoint].OutputValue, "postgres", 5432, username, password)
+
+	// Connect to postgres without a specific database to create the new one
+	db, err := sql.Open("postgres", adminEndpoint)
+	if err != nil {
+		return fmt.Errorf("failed to connect to postgres: %w", err)
+	}
+	defer db.Close()
+
+	// Create the database if it doesn't exist
+	if _, err := db.ExecContext(ctx, "CREATE DATABASE "+resourceID); err != nil {
+		// Ignore if database already exists
+		if !strings.Contains(err.Error(), "already exists") {
+			return fmt.Errorf("failed to create database: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func endpointToDSN(endpoint, database string, port int, username, password string) string {
+	urlEncodedPassword := url.QueryEscape(password)
+	return fmt.Sprintf("postgres://%s:%d/%s?user=%s&password=%s", endpoint, port, database, username, urlEncodedPassword)
```

Comment:
Would it be better to construct a `url.URL` here, then stringify it, to ensure all escaping is correct?